core/templates/hashfuncs.h

```diff
@@ -416,6 +300,28 @@ struct HashableHasher {
 	static _FORCE_INLINE_ uint32_t hash(const T &hashable) { return hashable.hash(); }
 };
 
+// Bitwise equality check. Faster than normal. Note: returns true if both values are NaN.
```

Comment:
This breaks the current contract of hash insertion and retrieval. We need NaN == NaN to be true.

A bitwise comparison is something else entirely, there are many NaN bit patterns in ieee 754. The is_nan check was added on purpose and cannot be removed.